### PR TITLE
codeintel: Add ownership and on-call sections

### PR DESCRIPTION
**Pull request by efritz**

### CODE REVIEW

**Review by aidaeology — APPROVED** [review]
Super LGTM ✅

File: handbook/engineering/code-intelligence/index.md [hunk 1]
```diff
@@ -123,6 +123,35 @@ Our team is growing and to help our new teammates have the best onboarding exper
...
+  - [sourcegraph/codeintelutils](https://github.com/sourcegraph/codeintelutils)
+  - [sourcegraph/code-intel-extensions](https://github.com/sourcegraph/code-intel-extensions)
+  - [sourcegraph/codeintellify](https://github.com/sourcegraph/codeintellify)
+  - [sourcegraph/lsif-go](https://github.com/sourcegraph/lsif-go)
+  - [sourcegraph/lsif-node](https://github.com/sourcegraph/lsif-node)
+  - [sourcegraph/lsif-java](https://github.com/sourcegraph/lsif-java)
+  - [sourcegraph/lsif-clang](https://github.com/sourcegraph/lsif-clang)
+  - [sourcegraph/lsif-test](https://github.com/sourcegraph/lsif-test)
+
+- BuildKite builds:
```
Comment (nicksnyder) [inline handbook/engineering/code-intelligence/index.md:144]:
```suggestion
- Buildkite builds:
```

**Review by nicksnyder — APPROVED** [review]

File: handbook/engineering/code-intelligence/index.md (hunk 1 above)
Comment (efritz) [reply handbook/engineering/code-intelligence/index.md:144]: 🙈 

Will open a PR to fix.

**Comment by nicksnyder** [comment]
fixes https://github.com/sourcegraph/sourcegraph/issues/13379

File: handbook/engineering/code-intelligence/index.md (hunk 1 above)
Comment (efritz) [reply handbook/engineering/code-intelligence/index.md:144]: https://github.com/sourcegraph/about/compare/fix-buildkite-casing?expand=1

**Pull request merged**